### DISCUSSION TITLE
`@image-url` from a `property<string>`

### DISCUSSION THREAD

**Discussion by microhobby** [slint-ui/slint#3437] [Q&A] [answered]
Should be possible to do something like:

```
root.imagepath ="images/nicolascage.png";
logo = @image-url("\{root.imagepath}");
```
This is returning `Syntax error: expected ')'`

Is there a way to set the string of the `@image-url` from a property?

**Comment by tronical** [comment, marked as answer]
Hi! You can use the C++ and Rust APIs of `slint::Image` to load an image at runtime from a path, and then set the image property.

That means that you can correctly compute the absolute path at runtime and handle errors if the file does not exist.

In theory we could allow arbitrary strings at runtime, but I feel that should use a function instead of a macro (`@image-url`), we would have to decide how to handle relative paths, we might need API to construct paths, and introduce features to make error handling ergonomic. That would seem like a direction to me that's closer to business logic, which I think we may want to minimize in .Slint files, also to keep them toolable.
